src/OpenTelemetry.Api/Context/Propagation/B3Propagator.cs

```diff
@@ -40,14 +40,19 @@ public sealed class B3Propagator : IPropagator
         // ActivityTraceId.SIZE hex characters (8-bytes traceId) in the past.
         internal const string UpperTraceId = "0000000000000000";
 
-        // Sampled value via the X_B3_SAMPLED header.
+        // Sampled values via the X_B3_SAMPLED header.
         internal const string SampledValue = "1";
 
+        // Some old zipkin implementations may send true/false for the sampled header. Only use this for checking incoming values.
+        internal const string LegacySampledValue = "true";
+
         // "Debug" sampled value.
         internal const string FlagsValue = "1";
 
         private static readonly HashSet<string> AllFields = new HashSet<string>() { XB3TraceId, XB3SpanId, XB3ParentSpanId, XB3Sampled, XB3Flags };
 
+        private static readonly HashSet<string> SampledValues = new HashSet<string>() { SampledValue, LegacySampledValue };
```

Comment:
Should we add in the hashset constructor something like:
new HashSet<string>(StringComparer.OrdinalIgnoreCase)

What do you think?